source/_patterns/03-templates/content-grid/content-grid.twig

```diff
@@ -0,0 +1,46 @@
+{%- set attributes = attributes|default({})|merge({class: attributes.class|default([])|merge(['content-grid']) }) -%}
+
+{%- set hasSecondary = false -%}
+{%- for area in content if area.area|default('full') == 'secondary' -%}
+  {%- set hasSecondary = true -%}
+{%- endfor -%}
+
+{%- if hasSecondary -%}
+  {%- set attributes = attributes|merge({class: attributes.class|merge(['content-grid--has-secondary']) }) -%}
+{%- endif -%}
+
+<{{ element|default('div') }} {%- include 'atoms-attributes' %}>
```

Comment:
Not a huge fan of making the element customisable. Only other option is just to rely on `role="main"`.